cli/cli-client/src/main/java/org/infinispan/cli/commands/Migrate.java

```diff
@@ -0,0 +1,261 @@
+
+   @CommandDefinition(name = Migrate.ClusterSynchronize.CMD, description = "Synchronizes data from a source to a target cluster")
+   public static class ClusterSynchronize extends CliCommand {
+      public static final String CMD = "synchronize";
+      public static final String READ_BATCH = "read-batch";
+      public static final String THREADS = "threads";
+
+      @Option(completer = CacheCompleter.class, shortName = 'c')
+      String cache;
+
+      @Option(shortName = 'h', hasValue = false, overrideRequired = true)
+      protected boolean help;
+
+      @Option(shortName = 'b', description = "The amount of entries to process in a batch")
+      Integer readBatch;
+
+      @Option(shortName = 't', description = "The number of threads to use. Defaults to the number of cores on the server")
+      Integer threads;
+
+      @Override
+      public boolean isHelp() {
+         return help;
+      }
+
+      @Override
+      protected CommandResult exec(ContextAwareCommandInvocation invocation) throws CommandException {
+         CommandInputLine cmd = new CommandInputLine(Migrate.CMD)
+               .arg(Migrate.TYPE, Migrate.Cluster.CMD)
+               .arg(Migrate.SUBTYPE, Migrate.ClusterSynchronize.CMD)
+               .option(READ_BATCH, readBatch)
+               .option(THREADS, threads)
+               .option(CACHE, cache);
+         return invocation.execute(cmd);
+      }
+   }
+
+   @GroupCommandDefinition(name = Migrate.Store.CMD, description = "Performs store migration", groupCommands = {Migrate.StoreFile.class}, activator = DisabledActivator.class)
+   public class Store extends CliCommand {
+
+      public static final String CMD = "store";
+
+      @Option(shortName = 'h', hasValue = false, overrideRequired = true)
+      protected boolean help;
+
+      @Override
+      public boolean isHelp() {
+         return help;
+      }
+
+      @Override
+      public CommandResult exec(ContextAwareCommandInvocation invocation) {
+         // This command serves only to wrap the sub-commands
+         invocation.println(invocation.getHelpInfo());
+         return CommandResult.FAILURE;
+      }
+   }
+
+   @CommandDefinition(name = Migrate.StoreFile.CMD, description = "Migrates a file store")
+   public static class StoreFile extends CliCommand {
+      public static final String CMD = "file";
+
+      @Option(completer = FileOptionCompleter.class, name = "source-location", required = true, description = "The location of the source file store.")
+      Resource sourceLocation;
+
+      @Option(completer = FileOptionCompleter.class, name = "target-location", required = true, description = "The location of the target file store.")
+      Resource targetLocation;
+
+      @Option(name = "source-version", required = true, description = "The source version.")
+      String sourceVersion;
+
+      @Option(name = "target-version", description = "The target version. Defaults to the version of this CLI.")
+      String targetVersion;
```

Comment:
The migrator utilises the current Infinispan version to write to the target store, so this isn't really configurable, it will just be the current Infinispan version associated with the CLI.